pkg/http/server.go

```diff
@@ -81,6 +91,48 @@ func (s *server) setupListener(opts Opts) error {
 	return nil
 }
 
+func setupUnixSocketListener(networkType string, address string) (net.Listener, error) {
+	socketOpts := strings.Split(address, ",")
+	if len(socketOpts) < 2 {
+		return net.Listen(networkType, address)
+	}
+
+	socketPath := socketOpts[0]
+
+	// must set umask to find out the previous value, so we set-store-reset it
+	currentUmask := syscall.Umask(0o777)
+	syscall.Umask(currentUmask)
+	socketMode := os.FileMode(currentUmask)
```

Comment:
It is almost never desirable to have the socket mode depend on the umask in the first place. Usually you want the socket to always have mode `660` allowing the owner and group to use the IPC.

It is of course always good if this can be overwritten (and I see that this PR also allows that), but a sensible default value would never depend on the process umask.

Reply:
I see, for normal files I would definitely agree with the other commenter. But the case of socket IPC is also special in that regard and should explicitly not depend on the umask. Even with the "standard" umask of 022 this would cause sockets to not be accessible by the group, which kind of defeats the whole purpose of IPC.